openrouteservice/src/main/java/org/heigit/ors/isochrones/GraphEdgeMapFinder.java

```diff
@@ -20,57 +20,90 @@
+        queryResults.add(res);
+        QueryGraph queryGraph = new QueryGraph(graph);
+        queryGraph.lookup(queryResults);
 
-		if (fromId == -1)
-			throw new InternalServerException(IsochronesErrorCodes.UNKNOWN, "The closest node is null.");
-	
-		Weighting weighting =  parameters.getRangeType() == TravelRangeType.TIME ?  new FastestWeighting(encoder) : new DistanceWeighting(encoder);
+        GHPoint3D snappedPosition = res.getSnappedPoint();
 
-		// IMPORTANT: It only works with TraversalMode.NODE_BASED.
-		DijkstraCostCondition dijkstraAlg = new DijkstraCostCondition(queryGraph, weighting, parameters.getMaximumRange(), parameters.getReverseDirection(),
-				TraversalMode.NODE_BASED);
-		dijkstraAlg.setEdgeFilter(edgeFilter);
-		dijkstraAlg.calcPath(fromId, Integer.MIN_VALUE);
+        int fromId = res.getClosestNode();
 
-		IntObjectMap<SPTEntry> edgeMap = dijkstraAlg.getMap();
-       return new AccessibilityMap(edgeMap, dijkstraAlg.getCurrentEdge(), snappedPosition);
-	}
+        if (fromId == -1)
+            throw new InternalServerException(IsochronesErrorCodes.UNKNOWN, "The closest node is null.");
+        Weighting weighting = createWeighting(parameters, encoder);
+
+        if (parameters.isTimeDependent()) {
+            //Time-dependent means traffic dependent for isochrones (for now)
+            TrafficSpeedCalculator trafficSpeedCalculator = new TrafficSpeedCalculator(weighting.getSpeedCalculator());
+            trafficSpeedCalculator.init(graph, encoder);
+            weighting.setSpeedCalculator(trafficSpeedCalculator);
+            if (HelperORS.getTurnCostExtensions(graph.getExtension()) != null)
+                weighting = new TurnWeighting(weighting, HelperORS.getTurnCostExtensions(graph.getExtension()));
+            TDDijkstraCostCondition tdDijkstraCostCondition = new TDDijkstraCostCondition(queryGraph, weighting, parameters.getMaximumRange(), parameters.getReverseDirection(),
+                    TraversalMode.NODE_BASED);
+            tdDijkstraCostCondition.setEdgeFilter(edgeFilter);
+            //Time is defined to be in UTC + 1 because original implementation was for German traffic data
+            //If changed, this needs to be adapted in the traffic storage, too
+            ZonedDateTime zdt = parameters.getRouteParameters().getDeparture().atZone(ZoneId.of("Europe/Berlin"));
+            trafficSpeedCalculator.setZonedDateTime(zdt);
+            int toId = parameters.getReverseDirection() ? fromId : Integer.MIN_VALUE;
+            fromId = parameters.getReverseDirection() ? Integer.MIN_VALUE : fromId;
+            tdDijkstraCostCondition.calcPath(fromId, toId, zdt.toInstant().toEpochMilli());
+            IntObjectMap<SPTEntry> edgeMap = tdDijkstraCostCondition.getMap();
+            return new AccessibilityMap(edgeMap, tdDijkstraCostCondition.getCurrentEdge(), snappedPosition);
+        } else {
+            // IMPORTANT: It only works with TraversalMode.NODE_BASED.
```

Comment:
There is no variable here to `assert`. It is a comment explaining why a parameter in the following constructor is set the way it is. Not new code either, but that doesn't really matter.

Reply:
O.K., I see. Maybe you could at least add an explanation why this is so?

Reply:
Not really, because this is not my code. It was written by Max I guess. And just from looking at it, I do not know why it only supports node based. Extends Dijkstra, and the extensions do not seem like they necessarily need to be node based to me. But I know as much as anyone else about that.

Reply:
Ok, then I suggest to would add a comment
`// TODO: understand reason for TraversalMode restriction`